### ISSUE TITLE
Docker - Upgrade to 3.4.0 leads to broken contacts view

### ISSUE THREAD

**Issue by Pamalosebi** [monicahq/monica#5684]
**Describe the bug**

Hi
I am using Docker to run Monica. I usually have not much trouble upgrading to a new version.
But since 3.4.0 I get the following message while opening any contact:
> Base table or view not found: 1146 Table 'monica.job_batches' doesn't exist

I upgraded successfully to 3.3.1. I also am able to downgrade again to 3.3.1.
The Issue really just appears using 3.4.0.

**Screenshots**
![image](https://user-images.githubusercontent.com/49550055/140640878-f2b5f8dd-038f-4ab5-8d8e-e2afe5eb79ec.png)

**Which version are you using:**
 - A server I maintain myself _(if so, please indicate your current version of Monica)_ -> 3.3.1 (since not able to upgrade properly)

**Comment by asbiin** [comment]
Looks like the migration didn't run. However, this is handle by the docker image, so if you have more logs that might be useful.


Try 
- restore your database backup
- run `php artisan migration` manually

**Comment by Pamalosebi** [comment]
Hey,

thank you for your reply! That did work.

So, what I did in case someone else has that problem in the future:
- connect to your container: `docker container exec -it <exact-name-of-your-container> /bin/bash`
- execute: `php artisan migrate` -> type "yes" and enter

Then everything was fine again.

**Comment by e-edgren** [comment]
I have the same issue and when I ran `php artisan migrate` and typed `yes`, I got the following issue:

![image](https://user-images.githubusercontent.com/53869451/142089161-213d8c90-e5c6-4a48-a2d5-b0be02f0f8e1.png)

What should I do?

**Comment by Pamalosebi** [comment]
When the migration doesn't work in the given way you might want ask the devs.
I am not sure fi you want to open a new Issue or if this one should be reopened.

**Comment by phylor** [comment]
@e-edgren I had the exact same problem as you. It seems that the migration `database/migrations/2020_04_24_212138_update_amount_format.php` only ran partially. I once heard that mysql does not support transactions regarding changes in the database schema. If that is still the case, I assume the migration changed the table schema, but then failed on the migration of the data. When rerunning, the schema changes can not be applied and the migration fails.

I used the following migration instead:

<details><summary>database/migrations/2020_04_24_212138_update_amount_format.php</summary>

```php
<?php

use App\Helpers\MoneyHelper;
use App\Models\Account\Account;
use Illuminate\Support\Facades\DB;
use Money\Currencies\ISOCurrencies;
use Money\Currency as MoneyCurrency;
use Illuminate\Support\Facades\Schema;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Eloquent\ModelNotFoundException;

class UpdateAmountFormat extends Migration
{
    /**
     * Run the migrations.
     *
     * @return void
     */
    public function up()
    {
        $this->fixDebts();
        $this->fixGifts();
    }

    private function fixDebts()
    {
        Schema::table('debts', function (Blueprint $table) {
            $table->decimal('amount', 13, 2)->change();
            //$table->unsignedInteger('currency_id')->after('amount')->nullable();
            //$table->foreign('currency_id')->references('id')->on('currencies')->onDelete('set null');
        });

        DB::table('debts')
            ->orderBy('id')
            ->chunk(500, function ($debts) {
                foreach ($debts as $debt) {
                    try {
                        $account = Account::findOrFail($debt->account_id);
                        $user = $account->users()->firstOrFail();
                    } catch (ModelNotFoundException $e) {
                        continue;
                    }

                    DB::update('update debts set amount = ?, currency_id = ? where id = ?', [
                        floatval($debt->amount) * self::unitAdjustment($user->currency),
                        $user->currency_id,
                        $debt->id,
                    ]);
                }
            });

        Schema::table('debts', function (Blueprint $table) {
            $table->integer('amount')->change();
        });
    }

    private function fixGifts()
    {
        Schema::table('gifts', function (Blueprint $table) {
            //$table->unsignedInteger('currency_id')->after('amount')->nullable();
            //$table->foreign('currency_id')->references('id')->on('currencies')->onDelete('set null');
        });

        DB::table('gifts')
            ->where('amount', '!=', 'null')
            ->orderBy('id')
            ->chunk(500, function ($gifts) {
                foreach ($gifts as $gift) {
                    try {
                        $account = Account::findOrFail($gift->account_id);
                        $user = $account->users()->firstOrFail();
                    } catch (ModelNotFoundException $e) {
                        continue;
                    }

                    DB::update('update gifts set amount = ?, currency_id = ? where id = ?', [
                        floatval($gift->amount) * self::unitAdjustment($user->currency),
                        $user->currency_id,
                        $gift->id,
                    ]);
                }
            });

        Schema::table('gifts', function (Blueprint $table) {
            $table->integer('amount')->change()->nullable();
            //$table->renameColumn('value', 'amount');
        });
    }

    /**
     * Get unit adjustement value for the currency.
     *
     * @param  \App\Models\Settings\Currency|int|null  $currency
     * @return int
     */
    private static function unitAdjustment($currency): int
    {
        $currency = MoneyHelper::getCurrency($currency);

        if (! $currency) {
            return 100;
        }

        $moneyCurrency = new MoneyCurrency($currency->iso);
        $currencies = new ISOCurrencies();

        return (int) pow(10, $currencies->subunitFor($moneyCurrency));
    }
}
```
</details>

Basically, I removed most of the schema migrations and renamed a couple of occurrences from `value` to `amount`. Use on your own risk of course 🤞
